### ISSUE TITLE
Smooth shows large  biases when numberOfPoints is low

### ISSUE THREAD

**Issue by veillette** [phetsims/calculus-grapher#293]
The blue and pink curve started the same, but I applied the smooth button to the blue curve. 

Below, this looks right for 400 points
http://localhost:63342/phetsims/calculus-grapher/calculus-grapher_en.html?brand=phet&ea&debugger&dev&numberOfPoints=400
![Calculus Grapher screenshot (46)](https://user-images.githubusercontent.com/7633344/225712671-1d3e2240-7a5c-4bd9-8bb8-8217a5ae1a3b.png)

The smoothing process, should be a weighted average, but should not biased the curve.
Notice that the centers don't match anymore below (Below numberOfPoints=100).

http://localhost:63342/phetsims/calculus-grapher/calculus-grapher_en.html?brand=phet&ea&debugger&dev&numberOfPoints=100
![Calculus Grapher screenshot (48)](https://user-images.githubusercontent.com/7633344/225715747-63aafaaa-bc5a-479a-a2eb-c8cc74cd8fd0.png)

This becomes very apparent when the number of points is ~100. It looks fine for 400 points. 
We should investigate what is going, and if it is worth fixing.

**Comment by veillette** [comment]
I understand what is  going on. This is the result of the interplay between the discretization of weighted sum average  and the discretization of the curve itself. This phenomenon becomes only apparent then when we reduced the number of points, such that the number of points on the left and on the right becomes unequal.

To remedy to this, we can explicitly fold our sum, such that we guarantee by construct that for every point that is pick to the left, we pick one to the right.  This is easy do to. Unfortunately, this makes the code less readable  but I'll add more comments/documentation to compensate for that.

**Comment by veillette** [comment]
This was added in the commit above.   In summary, we folded a sum of f(x) that would go from -x' to x' into a sum from that goes from 0 to x', but summing f(x') and f(-x). Special care is taken that we don't double count f(0) in this process.

I have added a lot of documentation to it, which is useful anyway as the code review mentioned that our documentation of these mathematical processes were scant..
  
I tested that it is working properly and that the smoothing does not show a bias.
